services/provider/server/server.go

```diff
@@ -410,20 +416,31 @@ func validateTicket(ticket string, pubKey *rsa.PublicKey) error {
 
 	message, err := base64.StdEncoding.DecodeString(ticketArr[0])
 	if err != nil {
-		return fmt.Errorf("failed to decode payload. %v", err)
+		return fmt.Errorf("failed to decode onboarding ticket: %v", err)
+	}
+
+	var ticketData onboardingTicket
+	err = json.Unmarshal(message, &ticketData)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal onboarding ticket message. %v", err)
 	}
+
 	signature, err := base64.StdEncoding.DecodeString(ticketArr[1])
 	if err != nil {
-		return fmt.Errorf("failed to decode signature. %v", err)
+		return fmt.Errorf("failed to decode onboarding ticket %s signature: %v", ticketData.ID, err)
 	}
 
-	hash := md5.Sum(message)
+	hash := sha256.Sum256(message)
 	err = rsa.VerifyPKCS1v15(pubKey, crypto.MD5, hash[:], signature)
```

Comment:
As seen here: https://github.com/jarrpa/ocs-operator/blob/ticketgen/hack/ticketgen/ticketcheck.go#L57
```suggestion
	err = rsa.VerifyPKCS1v15(pubKey, crypto.SHA256, hash[:], signature)
```